ecs-init/cache/dependencies.go

```diff
@@ -26,15 +26,103 @@ import (
 	"os"
 	"path/filepath"
 
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/s3"
 	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	log "github.com/cihub/seelog"
+	"github.com/pkg/errors"
 )
 
-// s3Downloader captures the only method used from the s3 client
-type s3Downloader interface {
+// s3API captures the only method used from the s3 package
+type s3API interface {
 	Download(w io.WriterAt, input *s3.GetObjectInput, options ...func(*s3manager.Downloader)) (n int64, err error)
 }
 
+type s3BucketDownloaderAPI interface {
+	download(fileName, cacheDir string, fs fileSystem) (string, error)
+}
+
+// s3BucketDownloader wraps a bucket together with a downloader that can download from it
+type s3BucketDownloader struct {
+	bucket string
+	region string
+	s3     s3API
+}
+
+func newS3BucketDownloader(region, bucketName string) (*s3BucketDownloader, error) {
+	session, err := session.NewSession(&aws.Config{
+		Credentials: credentials.AnonymousCredentials,
+		Region:      aws.String(region),
+	})
+	if err != nil {
+		log.Debugf("Got error when initializing downloader in region %s", region)
```

Comment:
Same as above. Can you please wrap the original error with something like `errors.Wrapf(err, "failed to initialize downloader in region %s", region)`?